File: include/quick/libqquickgraphicsconfiguration.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const std = @import("std");

/// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html)
pub const QQuickGraphicsConfiguration = extern struct {
    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html)
    ///
    /// The pointer to the underlying Qt C++ object
    ///
    ptr: QtC.QQuickGraphicsConfiguration,

    pub const _is_QQuickGraphicsConfiguration = {};

    /// ### DEPRECATED: Use `new` instead
    ///
    pub const New = new;

    /// Allocate a new QQuickGraphicsConfiguration object in C++ memory
    ///
    pub fn new() QQuickGraphicsConfiguration {
        return .{ .ptr = qtc.QQuickGraphicsConfiguration_new() };
    }

    /// ### DEPRECATED: Use `new2` instead
    ///
    pub const New2 = new2;

    /// Allocate a new QQuickGraphicsConfiguration object in C++ memory
    ///
    /// ## Parameter(s):
    ///
    /// ` other: QQuickGraphicsConfiguration `
    ///
    pub fn new2(other: anytype) QQuickGraphicsConfiguration {
        comptime _ = @TypeOf(other)._is_QQuickGraphicsConfiguration;
        return .{ .ptr = qtc.QQuickGraphicsConfiguration_new2(@ptrCast(other.ptr)) };
    }

    /// ### DEPRECATED: Use `operatorAssign` instead
    ///
    pub const OperatorAssign = operatorAssign;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#operator-eq)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    /// ` other: QQuickGraphicsConfiguration `
    ///
    pub fn operatorAssign(self: QQuickGraphicsConfiguration, other: anytype) void {
        comptime _ = @TypeOf(other)._is_QQuickGraphicsConfiguration;
        qtc.QQuickGraphicsConfiguration_OperatorAssign(@ptrCast(self.ptr), @ptrCast(other.ptr));
    }

    /// ### DEPRECATED: Use `preferredInstanceExtensions` instead
    ///
    pub const PreferredInstanceExtensions = preferredInstanceExtensions;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#preferredInstanceExtensions)
    ///
    /// ## Parameter(s):
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    pub fn preferredInstanceExtensions(allocator: std.mem.Allocator) [][]u8 {
        const _arr: qtc.libqt_list = qtc.QQuickGraphicsConfiguration_PreferredInstanceExtensions();
        var _str: [*]qtc.libqt_string = @ptrCast(@alignCast(_arr.data));
        defer {
            for (0.._arr.len) |i|
                qtc.libqt_string_free(@ptrCast(&_str[i]));
            qtc.libqt_free(_arr.data);
        }
        const _ret = allocator.alloc([]u8, _arr.len) catch @panic("QQuickGraphicsConfiguration.preferredInstanceExtensions: Memory allocation failed");
        for (0.._arr.len) |i| {
            const _data_val = _str[i];
            const _buf = allocator.alloc(u8, _data_val.len) catch @panic("QQuickGraphicsConfiguration.preferredInstanceExtensions: Memory allocation failed");
            @memcpy(_buf, _data_val.data[0.._data_val.len]);
            _ret[i] = _buf;
        }
        return _ret;
    }

    /// ### DEPRECATED: Use `setDeviceExtensions` instead
    ///
    pub const SetDeviceExtensions = setDeviceExtensions;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#setDeviceExtensions)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    /// ` extensions: [][]u8 `
    ///
    pub fn setDeviceExtensions(self: QQuickGraphicsConfiguration, allocator: std.mem.Allocator, extensions: [][]u8) void {
        const extensions_arr = allocator.alloc(qtc.libqt_string, extensions.len) catch @panic("QQuickGraphicsConfiguration.setDeviceExtensions: Memory allocation failed");
        defer allocator.free(extensions_arr);
        for (extensions, 0..extensions.len) |str_item, i|
            extensions_arr[i] = .{
                .len = str_item.len,
                .data = str_item.ptr,
            };
        const extensions_list = qtc.libqt_list{
            .len = extensions.len,
            .data = extensions_arr.ptr,
        };
        qtc.QQuickGraphicsConfiguration_SetDeviceExtensions(@ptrCast(self.ptr), extensions_list);
    }

    /// ### DEPRECATED: Use `deviceExtensions` instead
    ///
    pub const DeviceExtensions = deviceExtensions;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#deviceExtensions)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    pub fn deviceExtensions(self: QQuickGraphicsConfiguration, allocator: std.mem.Allocator) [][]u8 {
        const _arr: qtc.libqt_list = qtc.QQuickGraphicsConfiguration_DeviceExtensions(@ptrCast(self.ptr));
        var _str: [*]qtc.libqt_string = @ptrCast(@alignCast(_arr.data));
        defer {
            for (0.._arr.len) |i|
                qtc.libqt_string_free(@ptrCast(&_str[i]));
            qtc.libqt_free(_arr.data);
        }
        const _ret = allocator.alloc([]u8, _arr.len) catch @panic("QQuickGraphicsConfiguration.deviceExtensions: Memory allocation failed");
        for (0.._arr.len) |i| {
            const _data_val = _str[i];
            const _buf = allocator.alloc(u8, _data_val.len) catch @panic("QQuickGraphicsConfiguration.deviceExtensions: Memory allocation failed");
            @memcpy(_buf, _data_val.data[0.._data_val.len]);
            _ret[i] = _buf;
        }
        return _ret;
    }

    /// ### DEPRECATED: Use `setDepthBufferFor2D` instead
    ///
    pub const SetDepthBufferFor2D = setDepthBufferFor2D;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#setDepthBufferFor2D)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    /// ` enable: bool `
    ///
    pub fn setDepthBufferFor2D(self: QQuickGraphicsConfiguration, enable: bool) void {
        qtc.QQuickGraphicsConfiguration_SetDepthBufferFor2D(@ptrCast(self.ptr), enable);
    }

    /// ### DEPRECATED: Use `isDepthBufferEnabledFor2D` instead
    ///
    pub const IsDepthBufferEnabledFor2D = isDepthBufferEnabledFor2D;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#isDepthBufferEnabledFor2D)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    pub fn isDepthBufferEnabledFor2D(self: QQuickGraphicsConfiguration) bool {
        return qtc.QQuickGraphicsConfiguration_IsDepthBufferEnabledFor2D(@ptrCast(self.ptr));
    }

    /// ### DEPRECATED: Use `setDebugLayer` instead
    ///
    pub const SetDebugLayer = setDebugLayer;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#setDebugLayer)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    /// ` enable: bool `
    ///
    pub fn setDebugLayer(self: QQuickGraphicsConfiguration, enable: bool) void {
        qtc.QQuickGraphicsConfiguration_SetDebugLayer(@ptrCast(self.ptr), enable);
    }

    /// ### DEPRECATED: Use `isDebugLayerEnabled` instead
    ///
    pub const IsDebugLayerEnabled = isDebugLayerEnabled;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#isDebugLayerEnabled)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    pub fn isDebugLayerEnabled(self: QQuickGraphicsConfiguration) bool {
        return qtc.QQuickGraphicsConfiguration_IsDebugLayerEnabled(@ptrCast(self.ptr));
    }

    /// ### DEPRECATED: Use `setDebugMarkers` instead
    ///
    pub const SetDebugMarkers = setDebugMarkers;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#setDebugMarkers)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    /// ` enable: bool `
    ///
    pub fn setDebugMarkers(self: QQuickGraphicsConfiguration, enable: bool) void {
        qtc.QQuickGraphicsConfiguration_SetDebugMarkers(@ptrCast(self.ptr), enable);
    }

    /// ### DEPRECATED: Use `isDebugMarkersEnabled` instead
    ///
    pub const IsDebugMarkersEnabled = isDebugMarkersEnabled;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#isDebugMarkersEnabled)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    pub fn isDebugMarkersEnabled(self: QQuickGraphicsConfiguration) bool {
        return qtc.QQuickGraphicsConfiguration_IsDebugMarkersEnabled(@ptrCast(self.ptr));
    }

    /// ### DEPRECATED: Use `setTimestamps` instead
    ///
    pub const SetTimestamps = setTimestamps;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#setTimestamps)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    /// ` enable: bool `
    ///
    pub fn setTimestamps(self: QQuickGraphicsConfiguration, enable: bool) void {
        qtc.QQuickGraphicsConfiguration_SetTimestamps(@ptrCast(self.ptr), enable);
    }

    /// ### DEPRECATED: Use `timestampsEnabled` instead
    ///
    pub const TimestampsEnabled = timestampsEnabled;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#timestampsEnabled)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    pub fn timestampsEnabled(self: QQuickGraphicsConfiguration) bool {
        return qtc.QQuickGraphicsConfiguration_TimestampsEnabled(@ptrCast(self.ptr));
    }

    /// ### DEPRECATED: Use `setPreferSoftwareDevice` instead
    ///
    pub const SetPreferSoftwareDevice = setPreferSoftwareDevice;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#setPreferSoftwareDevice)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    /// ` enable: bool `
    ///
    pub fn setPreferSoftwareDevice(self: QQuickGraphicsConfiguration, enable: bool) void {
        qtc.QQuickGraphicsConfiguration_SetPreferSoftwareDevice(@ptrCast(self.ptr), enable);
    }

    /// ### DEPRECATED: Use `prefersSoftwareDevice` instead
    ///
    pub const PrefersSoftwareDevice = prefersSoftwareDevice;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#prefersSoftwareDevice)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    pub fn prefersSoftwareDevice(self: QQuickGraphicsConfiguration) bool {
        return qtc.QQuickGraphicsConfiguration_PrefersSoftwareDevice(@ptrCast(self.ptr));
    }

    /// ### DEPRECATED: Use `setAutomaticPipelineCache` instead
    ///
    pub const SetAutomaticPipelineCache = setAutomaticPipelineCache;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#setAutomaticPipelineCache)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    /// ` enable: bool `
    ///
    pub fn setAutomaticPipelineCache(self: QQuickGraphicsConfiguration, enable: bool) void {
        qtc.QQuickGraphicsConfiguration_SetAutomaticPipelineCache(@ptrCast(self.ptr), enable);
    }

    /// ### DEPRECATED: Use `isAutomaticPipelineCacheEnabled` instead
    ///
    pub const IsAutomaticPipelineCacheEnabled = isAutomaticPipelineCacheEnabled;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#isAutomaticPipelineCacheEnabled)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    pub fn isAutomaticPipelineCacheEnabled(self: QQuickGraphicsConfiguration) bool {
        return qtc.QQuickGraphicsConfiguration_IsAutomaticPipelineCacheEnabled(@ptrCast(self.ptr));
    }

    /// ### DEPRECATED: Use `setPipelineCacheSaveFile` instead
    ///
    pub const SetPipelineCacheSaveFile = setPipelineCacheSaveFile;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#setPipelineCacheSaveFile)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    /// ` filename: []const u8 `
    ///
    pub fn setPipelineCacheSaveFile(self: QQuickGraphicsConfiguration, filename: []const u8) void {
        const filename_str = qtc.libqt_string{
            .len = filename.len,
            .data = filename.ptr,
        };
        qtc.QQuickGraphicsConfiguration_SetPipelineCacheSaveFile(@ptrCast(self.ptr), filename_str);
    }

    /// ### DEPRECATED: Use `pipelineCacheSaveFile` instead
    ///
    pub const PipelineCacheSaveFile = pipelineCacheSaveFile;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#pipelineCacheSaveFile)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    pub fn pipelineCacheSaveFile(self: QQuickGraphicsConfiguration, allocator: std.mem.Allocator) []const u8 {
        var _str = qtc.QQuickGraphicsConfiguration_PipelineCacheSaveFile(@ptrCast(self.ptr));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("QQuickGraphicsConfiguration.pipelineCacheSaveFile: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// ### DEPRECATED: Use `setPipelineCacheLoadFile` instead
    ///
    pub const SetPipelineCacheLoadFile = setPipelineCacheLoadFile;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#setPipelineCacheLoadFile)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    /// ` filename: []const u8 `
    ///
    pub fn setPipelineCacheLoadFile(self: QQuickGraphicsConfiguration, filename: []const u8) void {
        const filename_str = qtc.libqt_string{
            .len = filename.len,
            .data = filename.ptr,
        };
        qtc.QQuickGraphicsConfiguration_SetPipelineCacheLoadFile(@ptrCast(self.ptr), filename_str);
    }

    /// ### DEPRECATED: Use `pipelineCacheLoadFile` instead
    ///
    pub const PipelineCacheLoadFile = pipelineCacheLoadFile;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#pipelineCacheLoadFile)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    pub fn pipelineCacheLoadFile(self: QQuickGraphicsConfiguration, allocator: std.mem.Allocator) []const u8 {
        var _str = qtc.QQuickGraphicsConfiguration_PipelineCacheLoadFile(@ptrCast(self.ptr));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("QQuickGraphicsConfiguration.pipelineCacheLoadFile: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// ### DEPRECATED: Use `delete` instead
    ///
    pub const Delete = delete;

    /// ### [Upstream resources](https://doc.qt.io/qt-6/qquickgraphicsconfiguration.html#dtor.QQuickGraphicsConfiguration)
    ///
    /// Delete this object from C++ memory
    ///
    /// ## Parameter:
    ///
    /// ` self: QQuickGraphicsConfiguration `
    ///
    pub fn delete(self: QQuickGraphicsConfiguration) void {
        qtc.QQuickGraphicsConfiguration_Delete(@ptrCast(self.ptr));
    }
};

File: src/quick/libqquickgraphicsconfiguration.cpp
#include <QByteArray>
#include <QList>
#include <QQuickGraphicsConfiguration>
#include <QString>
#include <qquickgraphicsconfiguration.h>
#include "libqquickgraphicsconfiguration.h"
#include "libqquickgraphicsconfiguration.hxx"

QQuickGraphicsConfiguration* QQuickGraphicsConfiguration_new() {
    return new QQuickGraphicsConfiguration();
}

QQuickGraphicsConfiguration* QQuickGraphicsConfiguration_new2(const QQuickGraphicsConfiguration* other) {
    return new QQuickGraphicsConfiguration(*other);
}

void QQuickGraphicsConfiguration_OperatorAssign(QQuickGraphicsConfiguration* self, const QQuickGraphicsConfiguration* other) {
    self->operator=(*other);
}

libqt_list /* of libqt_string */ QQuickGraphicsConfiguration_PreferredInstanceExtensions() {
    QList<QByteArray> _ret = QQuickGraphicsConfiguration::preferredInstanceExtensions();
    // Convert QList<> from C++ memory to manually-managed C memory
    libqt_string* _arr = static_cast<libqt_string*>(malloc(sizeof(libqt_string) * (_ret.size())));
    for (qsizetype i = 0; i < _ret.size(); ++i) {
        QByteArray _lv_qb = _ret[i];
        libqt_string _lv_str;
        _lv_str.len = _lv_qb.length();
        _lv_str.data = static_cast<char*>(malloc(_lv_str.len));
        memcpy((void*)_lv_str.data, _lv_qb.data(), _lv_str.len);
        _arr[i] = _lv_str;
    }
    libqt_list _out;
    _out.len = _ret.size();
    _out.data = static_cast<void*>(_arr);
    return _out;
}

void QQuickGraphicsConfiguration_SetDeviceExtensions(QQuickGraphicsConfiguration* self, const libqt_list /* of libqt_string */ extensions) {
    QList<QByteArray> extensions_QList;
    extensions_QList.reserve(extensions.len);
    libqt_string* extensions_arr = static_cast<libqt_string*>(extensions.data);
    for (size_t i = 0; i < extensions.len; ++i) {
        QByteArray extensions_arr_i_QByteArray(extensions_arr[i].data, extensions_arr[i].len);
        extensions_QList.push_back(extensions_arr_i_QByteArray);
    }
    self->setDeviceExtensions(extensions_QList);
}

libqt_list /* of libqt_string */ QQuickGraphicsConfiguration_DeviceExtensions(const QQuickGraphicsConfiguration* self) {
    QList<QByteArray> _ret = self->deviceExtensions();
    // Convert QList<> from C++ memory to manually-managed C memory
    libqt_string* _arr = static_cast<libqt_string*>(malloc(sizeof(libqt_string) * (_ret.size())));
    for (qsizetype i = 0; i < _ret.size(); ++i) {
        QByteArray _lv_qb = _ret[i];
        libqt_string _lv_str;
        _lv_str.len = _lv_qb.length();
        _lv_str.data = static_cast<char*>(malloc(_lv_str.len));
        memcpy((void*)_lv_str.data, _lv_qb.data(), _lv_str.len);
        _arr[i] = _lv_str;
    }
    libqt_list _out;
    _out.len = _ret.size();
    _out.data = static_cast<void*>(_arr);
    return _out;
}

void QQuickGraphicsConfiguration_SetDepthBufferFor2D(QQuickGraphicsConfiguration* self, bool enable) {
    self->setDepthBufferFor2D(enable);
}

bool QQuickGraphicsConfiguration_IsDepthBufferEnabledFor2D(const QQuickGraphicsConfiguration* self) {
    return self->isDepthBufferEnabledFor2D();
}

void QQuickGraphicsConfiguration_SetDebugLayer(QQuickGraphicsConfiguration* self, bool enable) {
    self->setDebugLayer(enable);
}

bool QQuickGraphicsConfiguration_IsDebugLayerEnabled(const QQuickGraphicsConfiguration* self) {
    return self->isDebugLayerEnabled();
}

void QQuickGraphicsConfiguration_SetDebugMarkers(QQuickGraphicsConfiguration* self, bool enable) {
    self->setDebugMarkers(enable);
}

bool QQuickGraphicsConfiguration_IsDebugMarkersEnabled(const QQuickGraphicsConfiguration* self) {
    return self->isDebugMarkersEnabled();
}

void QQuickGraphicsConfiguration_SetTimestamps(QQuickGraphicsConfiguration* self, bool enable) {
    self->setTimestamps(enable);
}

bool QQuickGraphicsConfiguration_TimestampsEnabled(const QQuickGraphicsConfiguration* self) {
    return self->timestampsEnabled();
}

void QQuickGraphicsConfiguration_SetPreferSoftwareDevice(QQuickGraphicsConfiguration* self, bool enable) {
    self->setPreferSoftwareDevice(enable);
}

bool QQuickGraphicsConfiguration_PrefersSoftwareDevice(const QQuickGraphicsConfiguration* self) {
    return self->prefersSoftwareDevice();
}

void QQuickGraphicsConfiguration_SetAutomaticPipelineCache(QQuickGraphicsConfiguration* self, bool enable) {
    self->setAutomaticPipelineCache(enable);
}

bool QQuickGraphicsConfiguration_IsAutomaticPipelineCacheEnabled(const QQuickGraphicsConfiguration* self) {
    return self->isAutomaticPipelineCacheEnabled();
}

void QQuickGraphicsConfiguration_SetPipelineCacheSaveFile(QQuickGraphicsConfiguration* self, const libqt_string filename) {
    QString filename_QString = QString::fromUtf8(filename.data, filename.len);
    self->setPipelineCacheSaveFile(filename_QString);
}

libqt_string QQuickGraphicsConfiguration_PipelineCacheSaveFile(const QQuickGraphicsConfiguration* self) {
    auto _ret = self->pipelineCacheSaveFile();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void QQuickGraphicsConfiguration_SetPipelineCacheLoadFile(QQuickGraphicsConfiguration* self, const libqt_string filename) {
    QString filename_QString = QString::fromUtf8(filename.data, filename.len);
    self->setPipelineCacheLoadFile(filename_QString);
}

libqt_string QQuickGraphicsConfiguration_PipelineCacheLoadFile(const QQuickGraphicsConfiguration* self) {
    auto _ret = self->pipelineCacheLoadFile();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void QQuickGraphicsConfiguration_Delete(QQuickGraphicsConfiguration* self) {
    delete self;
}
